src/V3AstNodes.cpp

```diff
@@ -1283,6 +1283,28 @@ string AstBasicDType::prettyDTypeName() const {
 void AstNodeMath::dump(std::ostream& str) const { this->AstNode::dump(str); }
 void AstNodeUniop::dump(std::ostream& str) const { this->AstNodeMath::dump(str); }
 
+std::string AstStrength::name() const {
+    string strengthString;
+    switch (this->m_level) {
+    case HIGHZ: strengthString = "highz"; break;
+    case SMALL: strengthString = "small"; break;
+    case MEDIUM: strengthString = "medium"; break;
+    case WEAK: strengthString = "weak"; break;
+    case LARGE: strengthString = "large"; break;
+    case PULL: strengthString = "pull"; break;
+    case STRONG: strengthString = "strong"; break;
+    case SUPPLY: strengthString = "supply";
+    }
+    strengthString += val ? '1' : '0';
+    return strengthString;
+}
+
+AstStrengthSpec::AstStrengthSpec(FileLine* fl, AstStrength* strength0p, AstStrength* strength1p)
```

Comment:
AST constructors are all in the header file.

Reply:
Done